### ISSUE TITLE
Publish to crates.io

### ISSUE THREAD

**Issue by o0Ignition0o** [apollographql/router#491]
Let's consider publishing router and core to crates.io

**Comment by abernix** [comment]
Since we think we're going to juggle around some internal crates before we go to 1.0, we've decided that while for 1.0 itself we will want to publish _everything_ to Crates.io (to simplify depending on compiled-in plugins), but that **prior to 1.0** — we will still require users to depend on the `apollo-router` and `apollo-router-core` from Git/GitHub based on tags.

In terms of what we will publish, eventually, it would be inclusive of code that is currently in `apollo-router`, `apollo-router-core` and the package currently known as `router-bridge` (but presumably named as `apollo-router-bridge`)

**Comment by garypen** [comment]
We would need to update our release automation to publish the crate to crates.io. Tooling required. I've used: https://github.com/crate-ci/cargo-release in the past and it might be helpful. Worth evaluating at least.

**Comment by o0Ignition0o** [comment]
Hacking on https://github.com/apollographql/federation-rs/issues/123, marking this one as blocked until i m done

**Comment by o0Ignition0o** [comment]
It's also blocked on defer + gzip (depending on how @Geal and us choose to deal with it)

**Comment by Geal** [comment]
> It's also blocked on defer + gzip (depending on how @Geal and we choose to deal with it)

Context: https://github.com/apollographql/router/pull/1604

**Comment by SimonSapin** [comment]
Should be fine to add an `apollo-` prefix to the names, right?

**Comment by SimonSapin** [comment]
With the publication of that crate as https://crates.io/crates/introspector-gadget, https://github.com/apollographql/router/pull/1751 now clears that last blocker in that last commit.

**Comment by Geal** [comment]
Wow that crate name. I love it

**Comment by abernix** [comment]
Full credit for the name goes to @EverlastingBugstopper.

**Comment by SimonSapin** [comment]
> that last blocker

of course, you know how these things go…
(another PR incoming)

**Comment by abernix** [comment]
[We are now published as `apollo-router` on crates.io](https://crates.io/crates/apollo-router).  Thanks for all the hard work to make this happen!
